farmdata2/farmdata2_modules/resources/BannerComponent.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * A Vue component for a message banner that is displayed when a specific event (success/error/etc.) occurs.
+ * 
+ * <h3 class="subsection-title">data-cy attributes</h3>
+ * <table>
+ * <thead><tr><th>Value</th>        <th>Description</th></tr></thead>
+ * <tbody>
+ * <tr><td>banner-handler</td>   <td>div containing bootstrap alert banner</td></tr>
+ * <tr><td>banner-close</td>   <td>'X' button that closes the alert banner</td></tr>
+ * <tr><td>banner-message</td>   <td>The message displayed inside the alert banner</td></tr>
+ * </tbody>
+ * </table>
+ * 
+ * <div>
+ *  @vue-prop {Object} [message={"msg": "Hello, I am a banner alert.", "class": "alert alert-warning"}] - Contains the message and class for the alert banner
+ *  @vue-prop {Boolean} [visible=null] - Set to true to show the alert banner. False makes the banner invisible.
+ *  @vue-prop {Boolean} [timeout=null] - Sets whether or not the banner dismisses itself which disables the 'x' to close. Set to false by default.

```

Comment:
Should indicate the amount of time before the banner is dismissed here.

Reply:
I'll be sure to add that to the documentation.